internals/link_helpers.py

```diff
@@ -47,6 +47,12 @@
 
 URL_REGEX = re.compile(r'(https?://\S+)')
 
+def valid_url(url):
+    try:
+        result = urlparse(url)
+        return all([result.scheme, result.netloc, result.path])
```

Comment:
Does this work for `http://example.com` with no trailing slash?  I think that makes path an empty string which is considered false.